include/exadg/structure/spatial_discretization/operator.cpp

```diff
@@ -935,6 +936,7 @@ Operator<dim, Number>::solve_nonlinear(VectorType &       sol,
   // set inhomogeneous Dirichlet values in order to evaluate the nonlinear residual correctly
   elasticity_operator_nonlinear.set_time(time);
   elasticity_operator_nonlinear.set_inhomogeneous_boundary_values(sol);
+  affine_constraints_periodicity_and_hanging_nodes.distribute(sol);
```

Comment:
> Most changes are made after solving. This change is made before solving. Why?

see https://github.com/exadg/exadg/pull/651#issuecomment-1991169639
Here, the `sol` is then used in the Newton solver to update the linearization vector
https://github.com/richardschu/exadg/blob/c5f489f1085b78f951d53cc62f510028c74c38be/include/exadg/solvers_and_preconditioners/newton/newton_solver.h#L79
and evaluate the residual
https://github.com/richardschu/exadg/blob/c5f489f1085b78f951d53cc62f510028c74c38be/include/exadg/solvers_and_preconditioners/newton/newton_solver.h#L104

Otherwise we have the zero hanging node values in the linearization vector.
I was surprised to not see an impact on the nonlinear iteration count, but maybe the solution we are searching for (with zero values at hanging nodes) is still close enough to the "real" one with correct hanging node values.

Reply:
> Did you see the TODO a few lines below? This code that is currently commented, but code that we might want to take care of as well.

I think we should remove it, when coming up with something new here.